pydata_center/monitoring/tests/test_views_dashboard.py

```diff
@@ -102,3 +115,95 @@ def test_dashboard_displays_multiple_agents(
         assert 'vm-online-01' in content
         assert 'vm-offline-02' in content
         assert content.count('class="offline"') == 1
+
+    def test_dashboard_displays_tags_correctly(
+            self,
+            authenticated_client,
+            url
+    ):
+        """
+        Tests that agent tags are correctly rendered in the dashboard HTML.
+        """
+        ServerStatus.objects.create(
+            hostname='agent-with-tags',
+            ip='10.0.0.1',
+            uptime=100,
+            healthy=True,
+            timestamp=now(),
+            tags={'env': 'prod', 'role': 'api'}
+        )
+        ServerStatus.objects.create(
+            hostname='agent-no-tags',
+            ip='10.0.0.2',
+            uptime=200,
+            healthy=True,
+            timestamp=now(), tags={}
+        )
+
+        response = authenticated_client.get(url)
+        content = response.content.decode()
+
+        assert response.status_code == 200
+        assert '<span class="tag">env: prod</span>' in content
+        assert '<span class="tag">role: api</span>' in content
+
+        assert 'agent-no-tags' in content
+        assert '<td>agent-no-tags</td>' in content

```

Comment:
Current checks like `'<span class="tag">env: prod</span>'` in content are good but brittle. If HTML structure or spacing changes slightly, the test may fail unnecessarily.
Consider using `BeautifulSoup` to parse and assert on tag values more robustly:
```
from bs4 import BeautifulSoup

soup = BeautifulSoup(content, 'html.parser')
tags = {tag.text.strip() for tag in soup.select('span.tag')}
self.assertIn('env: prod', tags)
self.assertIn('role: api', tags)
```